linera-chain/src/manager.rs

```diff
@@ -404,7 +404,7 @@ where
         // We don't compare to `current_round` here: Non-validators must update their locked block
         // even if it is older than the current round. Validators will only sign in the current
         // round, though. (See `create_final_vote` below.)
-        if let Some(locked) = self.locked.get() {
+        if let Some(LockedBlock::Regular(locked)) = self.locked.get() {
```

Comment:
What if we have a `::Fast` block here? Should that be unreachable?

Reply:
Not unreachable, but if our locked block is in round `Fast`, then `new_round` will always be greater: `Fast` is the lowest round, and validated block certificates only exist in other rounds.

Reply:
OK. My point is - we're silently ignoring that variant here. I'd be more comfortable if we pattern match on both variants and handle them explicitly. For `Fast` variant maybe add some assertions/comments. The code is quite complex already.

Reply:
Done in https://github.com/linera-io/linera-protocol/pull/3140/commits/3876f9ad5dd59e4c6e4121e8b9078590fa627c81.

I just do the check in either case, even though if the locked block is fast it will be trivially satisfied.